### PR TITLE
Setup redux with dark mode state management

### PR DESCRIPTION
**Pull request by adindotdev**
Lots of boilerplate. Feel free to change the project structure of this in future iterations.

### CODE REVIEW

File: src/state/management/userSettings.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Initial state
+const INITIAL_STATE = {
+  isDarkModeEnabled: false,
```
Comment (ianmah) [inline src/state/management/userSettings.js:3]:
```suggestion
  isDarkModeEnabled: true,
```

**Review by ianmah — COMMENTED** [review]
Do you mind adding a usage example somewhere? Like an example function

File: src/state/management/userSettings.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// Types
+export const TOGGLE_DARK_MODE = 'TOGGLE_DARK_MODE';
+
+// Actions
+export const toggleDarkMode = (isDarkModeEnabled) => {
+  return {
+    type: TOGGLE_DARK_MODE,
+    isDarkModeEnabled: isDarkModeEnabled,
+  };
+};
```
Comment (ianmah) [inline src/state/management/userSettings.js:15]:
```suggestion
export const toggleDarkMode = () => {
  return {
    type: TOGGLE_DARK_MODE,
    isDarkModeEnabled: !isDarkModeEnabled,
  };
};
```

maybe toggle?

Comment (adindotdev) [reply src/state/management/userSettings.js:15]: I debated this for 10 seconds when making this. I think passing in a boolean is better for consistency with DB level (ie. default value comes from DB)

**Review by vandyliu — APPROVED** [review]
looks good

**Pull request merged**